scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -822,5 +828,200 @@
       (is (= [(make-job 1 2 2048) (make-job 2 1 1024)]
              (sched/filter-based-on-quota {test-user {:count 4, :cpus 20, :mem 6144}} user->usage queue))))))
 
+(deftest test-handle-resource-offers
+  (let [uri "datomic:mem://test-handle-resource-offers"
+        launched-offer-ids-atom (atom [])
+        launched-task-ids-atom (atom [])
+        driver (reify msched/SchedulerDriver
+                 (launch-tasks! [_ offer-id tasks]
+                   (swap! launched-offer-ids-atom conj (-> offer-id first :value))
+                   (swap! launched-task-ids-atom concat (map (fn [task]
+                                                               {:id (get-in task [:task-id :value])
+                                                                :cpus (->> task
```

Comment:
I'm using prime numbers for the cpus and validating their sum is what is matched. I'll do soemthing similar with `:mem`.